syntax = "proto3";

package platform.agent.v1;

import "platform/tool/v1/tool_call.proto";

message Block {
  string id = 1;
  int32 sequence = 2;
  oneof content {
    UserInput user_input = 3;
    platform.tool.v1.ToolCall tool_call = 4;
    AssistantMessage assistant_message = 5;
  }
}

message UserInput {
  string text = 1;
  repeated Attachment attachments = 2;
}

message AssistantMessage {
  string text = 1;
}

message Attachment {
  string id = 1;
  string filename = 2;
  string mime_type = 3;
  string url = 4;
}


